### ISSUE TITLE
About test Set5 have some error

### ISSUE THREAD

**Issue by echosum** [thstkdgus35/EDSR-PyTorch#163]
Hey, I download your predtrain model ,EDSRX2 to test Set5!. But have some pixel error
![image](https://user-images.githubusercontent.com/39319252/57343427-f1999f80-7175-11e9-9f09-80378804e13b.png)

And the PSNR of Set5 is 39.14. It seem big than your paper?

**Comment by thstkdgus35** [comment]
Hello.

Would you provide the test script you used? I think I can give you an answer with #164 

Thank you.

**Comment by echosum** [comment]
[code.zip](https://github.com/thstkdgus35/EDSR-PyTorch/files/3159958/code.zip)

**Comment by echosum** [comment]
Hi.
I have send my code. If u want to run this code. U just modify the test.py line80 for your model path, and data_fortest.py line17 for your benchmark path(Data folder structure is same with yous).

**Comment by thstkdgus35** [comment]
@echosum Sorry but for some personal reason, I cannot run your code right now.

I checked the code and thought that line 45, 47 and 48 of your ``test.py` may cause the problem.

This problem usually happens when the output image is not clipped well.

In other words, you need to check that pixel values of the quantized image are in [0, 255].

If not, you can simply resolve this problem by clipping them.

Please let me know if you cannot fix the problem after the solution :)

Thank you

**Comment by echosum** [comment]
`                hr = model(lr)
                hr = common.quantize(hr,255)
                print('range',torch.max(hr),torch.min(hr))
                hr_np = hr.cpu().data.numpy().astype('uint8').transpose(0, 2, 3, 1)
                print(np.max(hr_np),np.min(hr_np))
                img = Image.fromarray(hr_np[0],'RGB')
                img.save('results/%s'%batch['info'][0])`

**Comment by echosum** [comment]
Hi. 
In the code that I send to you. I have check the range of HR image is [0,255].  The PSNR is always different with the result in your paper. I always can't fix the problem by myself. I will appreciate you.
Thanks for your help.

**Comment by echosum** [comment]
Hi I try many ways to find the problem that PSNR is wrong. I think maybe the key in the input of model.

**Comment by echosum** [comment]
Hi， I have fix these problem. 
Thanks for your help

**Comment by Ethean** [comment]
@echosum hi, I downloaded the edsr_x4.pth and test it on the validation set of div2k and I got a pretty low psnr(19.7) and ssim(0.459). Then I test on set5 and the psnr still has a very low value(19.2). What could I miss?

**Comment by AbdulMoqeet** [comment]
@echosum  Make sure, you are loading the trained model. Just setting the `--pre_train download` will also serve the purpose.

**Comment by suxi-du** [comment]
> Hi， I have fix these problem. Thanks for your help

hey, how did u fix it?

**Comment by ptrahuang** [comment]
> > Hi， I have fix these problem. Thanks for your help
> 
> hey, how did u fix it?

hey ，how did u fix it ？
